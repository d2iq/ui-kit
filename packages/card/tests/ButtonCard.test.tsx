import React from "react";
import { mount } from "enzyme";
import { createSerializer } from "@emotion/jest";
import toJson from "enzyme-to-json";

import { ButtonCard } from "../";

expect.addSnapshotSerializer(createSerializer());

describe("ButtonCard", () => {
  it("renders default", () => {
    const component = mount(<ButtonCard>default</ButtonCard>);

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders as isActive", () => {
    const component = mount(<ButtonCard isActive={true}>default</ButtonCard>);

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders as disabled", () => {
    const component = mount(<ButtonCard disabled={true}>default</ButtonCard>);

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders as hasFocus", () => {
    const component = mount(<ButtonCard hasFocus={true}>default</ButtonCard>);

    expect(toJson(component)).toMatchSnapshot();
  });
  it("does not pass accessibility props if isInput is true", () => {
    const accessibilityProps = [
      "tabIndex",
      "role",
      "aria-disabled",
      "aria-pressed"
    ];
    const component = mount(<ButtonCard isInput={true}>default</ButtonCard>);
    const buttonCardNodeProps = component
      .find("[data-cy='buttonCard']")
      .props();
    const propKeys = Object.keys(buttonCardNodeProps).map(key => key);

    accessibilityProps.forEach(prop => {
      expect(propKeys.includes(prop)).toBe(false);
    });
    // expect(toJson(component)).toMatchSnapshot();
  });
});
