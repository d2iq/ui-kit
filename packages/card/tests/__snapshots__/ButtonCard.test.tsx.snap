// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCard renders as disabled 1`] = `
.emotion-0 {
  background-color: var(--themeBgPrimary, #FFFFFF);
  box-shadow: 0 0 0 1px var(--themeBorder, #DADDE2);
  border-radius: 6px;
  cursor: pointer;
  background-color: var(--themeBgDisabled, #E8EAED);
  color: var(--themeTextColorDisabled, #AEB0B4);
  cursor: auto;
}

.emotion-0>div {
  height: 100%;
}

.emotion-0:hover,
.emotion-0:focus,
.emotion-0:focus-within {
  box-shadow: 0 0 0 1px var(--themeBrandPrimary, #7D58FF);
  outline: 0;
}

.emotion-0:active {
  box-shadow: 0 0 0 2px var(--themeBrandPrimary, #7D58FF);
}

.emotion-0,
.emotion-0:hover,
.emotion-0:focus {
  box-shadow: 0 0 0 0 transparent;
}

.emotion-1 {
  padding: 16px;
}

<ButtonCard
  disabled={true}
  paddingSize="m"
>
  <div
    aria-disabled={true}
    className="emotion-0"
    data-cy="buttonCard"
    onKeyPress={[Function]}
    role="button"
    tabIndex={-1}
  >
    <div
      className="emotion-1"
    >
      default
    </div>
  </div>
</ButtonCard>
`;

exports[`ButtonCard renders as hasFocus 1`] = `
.emotion-0 {
  background-color: var(--themeBgPrimary, #FFFFFF);
  box-shadow: 0 0 0 1px var(--themeBorder, #DADDE2);
  border-radius: 6px;
  cursor: pointer;
  box-shadow: 0 0 0 1px var(--themeBrandPrimary, #7D58FF);
  outline: 0;
}

.emotion-0>div {
  height: 100%;
}

.emotion-0:hover,
.emotion-0:focus,
.emotion-0:focus-within {
  box-shadow: 0 0 0 1px var(--themeBrandPrimary, #7D58FF);
  outline: 0;
}

.emotion-0:active {
  box-shadow: 0 0 0 2px var(--themeBrandPrimary, #7D58FF);
}

.emotion-1 {
  padding: 16px;
}

<ButtonCard
  hasFocus={true}
  paddingSize="m"
>
  <div
    className="emotion-0"
    data-cy="buttonCard"
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="emotion-1"
    >
      default
    </div>
  </div>
</ButtonCard>
`;

exports[`ButtonCard renders as isActive 1`] = `
.emotion-0 {
  background-color: var(--themeBgPrimary, #FFFFFF);
  box-shadow: 0 0 0 1px var(--themeBorder, #DADDE2);
  border-radius: 6px;
  cursor: pointer;
}

.emotion-0>div {
  height: 100%;
}

.emotion-0:hover,
.emotion-0:focus,
.emotion-0:focus-within {
  box-shadow: 0 0 0 1px var(--themeBrandPrimary, #7D58FF);
  outline: 0;
}

.emotion-0:active {
  box-shadow: 0 0 0 2px var(--themeBrandPrimary, #7D58FF);
}

.emotion-0,
.emotion-0:hover,
.emotion-0:focus {
  box-shadow: 0 0 0 2px var(--themeBrandPrimary, #7D58FF);
}

.emotion-0:focus {
  background-color: var(--themeBgHover, #F7F8F9);
}

.emotion-1 {
  padding: 16px;
}

<ButtonCard
  isActive={true}
  paddingSize="m"
>
  <div
    aria-pressed={true}
    className="emotion-0"
    data-cy="buttonCard"
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="emotion-1"
    >
      default
    </div>
  </div>
</ButtonCard>
`;

exports[`ButtonCard renders default 1`] = `
.emotion-0 {
  background-color: var(--themeBgPrimary, #FFFFFF);
  box-shadow: 0 0 0 1px var(--themeBorder, #DADDE2);
  border-radius: 6px;
  cursor: pointer;
}

.emotion-0>div {
  height: 100%;
}

.emotion-0:hover,
.emotion-0:focus,
.emotion-0:focus-within {
  box-shadow: 0 0 0 1px var(--themeBrandPrimary, #7D58FF);
  outline: 0;
}

.emotion-0:active {
  box-shadow: 0 0 0 2px var(--themeBrandPrimary, #7D58FF);
}

.emotion-1 {
  padding: 16px;
}

<ButtonCard
  paddingSize="m"
>
  <div
    className="emotion-0"
    data-cy="buttonCard"
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="emotion-1"
    >
      default
    </div>
  </div>
</ButtonCard>
`;
